lib/api/src/rc_switch_user.cpp

```diff
@@ -0,0 +1,71 @@
+#include "irods/switch_user.h"
+
+#include "irods/plugins/api/switch_user_types.h"
+#include "irods/plugins/api/api_plugin_number.h"
+#include "irods/procApiRequest.h"
+#include "irods/rodsErrorTable.h"
+
+#include <cstdlib>
+#include <cstring>
+
+auto rc_switch_user(RcComm* _comm, const char* _username, const char* _zone) -> int
+{
+    // NOLINTNEXTLINE(readability-implicit-bool-conversion)
+    if (!_username || !_zone) {
+        return SYS_INVALID_INPUT_PARAM;
+    }
+
+    auto& client = _comm->clientUser;
+
+    // TODO
+    // Do nothing if the RcComm already represents the target user.
+    // Not doing this is good for benchmarking.
+    //
+    // Should we do this check at all?
+    // How do we know the RcComm and RsComm represent the same user?
+    //
+    //if (std::strcmp(_username, static_cast<const char*>(client.userName)) == 0 &&
+    //    std::strcmp(_zone, static_cast<const char*>(client.rodsZone)) == 0)
+    //{
+    //    return 0;
+    //}
+
+    // TODO
+    // Does the .irodsA file matter at all?
```

Comment:
Investigate whether this matters.

Reply:
The .irodsA file only matters when authenticating with the server. This API plugin is designed to be used only after authentication.